sql/core/src/test/scala/org/apache/spark/sql/execution/SubExprEliminationBenchmark.scala

```diff
@@ -108,11 +110,81 @@ object SubExprEliminationBenchmark extends SqlBasedBenchmark {
     }
   }
 
+  def withFilter(rowsNum: Int, numIters: Int): Unit = {
+    val benchmark = new Benchmark("from_json as subExpr in Filter", rowsNum, output = output)
+
+    withTempPath { path =>
+      prepareDataInfo(benchmark)
+      val numCols = 1000
+      val schema = writeWideRow(path.getAbsolutePath, rowsNum, numCols)
+
+      val predicate = (0 until numCols).map { idx =>
+        (from_json('value, schema).getField(s"col$idx") >= Literal(100000)).expr
+      }.asInstanceOf[Seq[Expression]].reduce(Or)
+
+      // We only benchmark subexpression performance under codegen/non-codegen, so disabling
+      // json optimization.
+      benchmark.addCase("subexpressionElimination off, codegen on", numIters) { _ =>
+        withSQLConf(
+          SQLConf.SUBEXPRESSION_ELIMINATION_ENABLED.key -> "false",
```

Comment:
Should we maybe better do this with a loop? e.g.)

```scala
Seq(("false", "true", "CODEGEN_ONLY", "false"), ...). foreach {
  case (
       subexpressionElimitationEnabled,
       ...  =>
     benchmark.addCase(s"${SQLConf.key} $subexpressionElimitationEnabled, ...", numIters) { _ =>
        withSQLConf(
          SQLConf.key -> subexpressionElimitationEnabled,
          ...
```

Reply:
I believe we can do this separately to refactor this whole file.

Reply:
Okay, I will refactor it.